app/assets/javascripts/Components/Modals/submission_file_upload_modal.jsx

```diff
@@ -16,17 +17,88 @@ class SubmissionFileUploadModal extends React.Component {
 
   onSubmit = event => {
     event.preventDefault();
-    this.props.onSubmit(this.state.newFiles, this.state.unzip);
+    this.props.onSubmit(this.state.newFiles, this.state.unzip, this.state.single_file_name);
   };
 
   handleFileUpload = event => {
-    this.setState({newFiles: event.target.files});
+    this.setState({newFiles: event.target.files, single_file_name: event.target.files[0].name});
   };
 
   toggleUnzip = () => {
     this.setState({unzip: !this.state.unzip});
   };
 
+  handleNameChange = event => {
+    this.setState({single_file_name: event.target.value});
+  };
+
+  fileRenameInputBox = () => {
+    let fileRenameInput;
+    let filesToShow;
+    if (this.props.uploadTarget) {
+      let filenames = this.props.requiredFiles.filter(
+        filename =>
+          filename.indexOf(this.props.uploadTarget) === 0 &&
+          filename.slice(this.props.uploadTarget.length).indexOf("/") === -1
```

Comment:
Use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes, and note that this method takes a `position` argument.